core/trino-spi/src/test/java/io/trino/spi/type/TestUnscaledDecimal128Arithmetic.java

```diff
@@ -347,16 +386,9 @@ public void testOverflows()
     public void testCompare()
     {
         assertCompare(unscaledDecimal(0), unscaledDecimal(0), 0);
-        assertCompare(negate(unscaledDecimal(0)), unscaledDecimal(0), 0);
```

Comment:
We should add an argument check in `io.trino.spi.type.UnscaledDecimal128Arithmetic#negate` so that it cannot be used in the future to introduce a negative zero.

However, since this function has one non-test usage only, we can maybe inline it? (follow up)

Reply:
I considered adding some checks to this method, but I treat it somewhat like an internal API (even though it's used by another class as well), so it's one of those methods that you use when you know what you're doing.

Reply:
i agree that internal code doesn't have -- and likely shouldn't have -- checks sprinkled eveywhere.
OTOH i am afraid we will soon forget about all this.

hopefully the  validate-on-return check may help us remember.

Reply:
IMO we could add code similar to code in `pack` for `negate`. `negate` is only used (apart from tests) in `io.trino.operator.scalar.MathFunctions.Abs#absLong`, so it's not that critical method.